odxtools/table.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2022 MBition GmbH
+
+import abc
+from dataclasses import dataclass
+from typing import Optional, List
+
+from .globals import logger
+
+
+class TableBase(abc.ABC):
+    """ Base class for all Tables."""
+
+    def __init__(self, id: str, short_name: str, key_dop_ref: str, long_name=None):
+        self.id = id
+        self.short_name = short_name
+        self.long_name = long_name
+        self.key_dop_ref = key_dop_ref
+
+
+@dataclass()
+class TableRow:
+    """This class represents a TABLE-ROW."""
+
+    id: str
+    short_name: str
+    long_name: str
+    key: int
+    structure_ref: str
```

Comment:
you should probably resolve references. the mechanism is to add a `_resolve_references(id_lookup)` method to all classes containing references and call it from the instancing class' `_respolve_references()`. in this case, you should add these method to `TableRow` and `Table` and call the latter one from  [DataDictionarySpec](https://github.com/mercedes-benz/odxtools/blob/main/odxtools/diagdatadictionaryspec.py#L75).

Reply:
Sure, I'll add the recommended approach.

Reply:
Hi @andlaus - I've added the functionality to resolve the references for TABLE and TABLE-ROWS. Please review. 

I'll review the templates and the test cases for tables in the next few days.